### DISCUSSION TITLE
Determining applied representations programmatically?

### DISCUSSION THREAD

**Discussion by whickman** [molstar/molstar#1141] [Q&A] [answered]
Hello! I'm attempting to determine applied representations programatically. It seems like I might be able to do this by looking at the properties and parents of `cells` (`plugin.state.data.cells`) to determine which are representations on the `StateObject`s I'm interested in, but I'm wondering if there's a better way?

I've got a use-case where I frequently need to remove portions of a structure and create a new merged structure, so am trying to keep user-applied preset representations (e.g. Gaussian Surface) consistent across the state change.

There's been some discussion around programatically changing representations (e.g. [here](https://github.com/molstar/molstar/discussions/301)), but I couldn't find anything in the discussions or code which answered the question above.

Thanks!

**Comment by dsehnal** [comment]
I don't know your business logic, but the state manager keeps the current representations automatically when you update a parent node in the tree. For a more complex interactions, you can use `plugin.state.data.selectQ` to query subtrees of nodes and re-apply the same transforms to new ones.

**Reply by whickman** [reply, marked as answer]
Thanks! I ended up doing something along those lines -- in my case, I'm applying the `Polymer and Ligand` preset to the merged structure, and so was able to use the tags created by that (`structure-component-static-polymer` and `structure-component-static-ligand`) to find the relevant transform subtrees across the two states (which otherwise have different `ref`s for pretty much everything) and then re-apply any relevant transforms to the updated state.

not sure if that's the most straightforward way, but it works -- thanks again!
